docs/current_docs/api/fs-filters.mdx

```diff
@@ -0,0 +1,308 @@
+
+Dagger offers pre- and post-call filtering to mitigate this problem and optimize how your directories are handled.
+
+## Benefits
+
+Filtering improves the performance of your Dagger Functions in two ways:
+
+- It reduces the size of the files being transferred from the host to the Dagger Engine, allowing the upload step to complete faster.
+- It ensures that minor unrelated changes in the source directory don't invalidate Dagger's build cache.
+- It enables different use-cases, such as setting up pipelines for monorepos.
+
+It is worth noting that Dagger already uses caching to optimize file uploads. Subsequent calls to a Dagger Function will only upload files that have changed since the preceding call. Filtering is an additional optimization that you can apply to improve the performance of your Dagger Function.
+
+## Pre-call filtering
+
+Pre-call filtering means that a directory is filtered before it's uploaded to the Dagger Engine container. This is useful for:
+
+- Monorepos containing multiple files and directories. Typically in a monorepo, your Dagger Function only operates on a subset (representing a specific component or feature) of your repository, so uploading the entire worktree on every call imposes a prohibitive cost.
+
+- Repositories containing binary objects. Binary objects, such as audio/video files, are usually extremely large in size and therefore take time to upload. If they're not directly relevant, you'll usually want your Dagger Function to ignore them.
+
+  :::tip
+  The `.git` directory is a good example of both these cases. It contains a lot of data, including large binary objects, and for projects with a long version history, it can sometimes be larger than your actual source code.
+  :::
+
+- Dependencies. If you're developing locally, you'll typically have your project dependencies installed locally as well: `node_modules` (Node.js), `.venv` (Python), `vendor` (PHP) and so on. When you call your Dagger Function locally, Dagger will upload all these installed dependencies as well. This is both bad practice and inefficient. Typically, you'll want your Dagger Function to ignore locally-installed dependencies and only operate on the project source code.
+
+:::note
+At the time of writing, Dagger does not read exclusion patterns from existing `.dockerignore`/`.gitignore` files. If you already use these files, you'll need to manually implement the same patterns in your Dagger Function.
+:::
+
+To implement a pre-call filter in your Dagger Function, add an `ignore` parameter to your `Directory` argument. The `ignore` parameter follows the `.gitignore` syntax: prefixing a path with `!` will actually include it instead of ignoring it.
```

Comment:
`!` does not technically include the file ; it negate a previous ignore.

For example: 
- `// +ignore=["!foo"]` -> Has 0 effect since nothing is ignore, `foo` would have been include
- `// +ignore=["**", "!foo"]` -> Will exclude everything except foo

Also not that the order of the argument actually matters, `["**", "!**"]` Will include everything but `["!**", "**"]` will actually exclude everything.
There's should be a mention somewhere of this.